cirq-google/cirq_google/api/v2/sweeps.py

```diff
@@ -55,14 +56,18 @@ def _recover_sweep_const(const_pb: run_context_pb2.ConstValue) -> Any:
 
 
 def sweep_to_proto(
-    sweep: cirq.Sweep, *, out: Optional[run_context_pb2.Sweep] = None
+    sweep: cirq.Sweep,
+    *,
+    out: Optional[run_context_pb2.Sweep] = None,
+    sweep_transformer: Callable[[cirq.Points | cirq.Linspace], cirq.Sweep] = lambda x: x,
```

Comment:
@senecameeks for typing you can do this

```py3
_SWEEP_T = TypeVar('_SWEEP_T', cirq.Points, cirq.Linspace)  # or just cirq.SingleSweep

Callable[[_SWEEP_T], _SWEEP_T]
```

this will remove the cast statements below

Reply:
this still causes mypy to fail, but also I prefer using `cirq.SingleSweep` in the call over type vars since IMO the former is more readable. So I'll keep as is

Reply:
the TypeVar tells mypy that the type of the output is the same as the type of the input